### PR TITLE
Create page for Python web scraping using requests

### PR DESCRIPTION
**Pull request by tlan4**
An introduction on how to use the Python Requests library to query data from websites, and then parse it using the BeautifulSoup4 library. Provides a walkthrough on how to install the required libraries, determine the request needed to scrape the desired data, and parse the content to extract the desired data from the response.

### CODE REVIEW

**Comment by chenbriann** [comment]
Hi, I’ve already made a wiki page on using requests and beautiful soup (see pull request #234 ) so unfortunately this pull request will create a conflict win the wiki.

**Comment by tlan4** [comment]
> Hi, I’ve already made a wiki page on using requests and beautiful soup (see pull request #234 ) so unfortunately this pull request will create a conflict in the wiki.

What if you focus on the BeautifulSoup part and I can write more about the requests portion?

**Comment by tlan4** [comment]
> > > Hi, I’ve already made a wiki page on using requests and beautiful soup (see pull request #234 ) so unfortunately this pull request will create a conflict in the wiki.
> > 
> > 
> > What if you focus on the BeautifulSoup part and I can write more about the requests portion?
> 
> I need to use the requests library for my guide to make sense. Maybe you could make your guide on using requests for a topic other than web scraping? Or using a different parser like lxml and write a guide for using XPath expressions with requests.

I was thinking about talking more in detail about stuff like POST and Session and Content-Type for Requests in web scraping, and I could just quickly mention bs4. And you can keep what you currently have for requests, and write more in detail about bs4.

**Comment by chenbriann** [comment]
> > > > Hi, I’ve already made a wiki page on using requests and beautiful soup (see pull request #234 ) so unfortunately this pull request will create a conflict in the wiki.
> > > 
> > > 
> > > What if you focus on the BeautifulSoup part and I can write more about the requests portion?
> > 
> > 
> > I need to use the requests library for my guide to make sense. Maybe you could make your guide on using requests for a topic other than web scraping? Or using a different parser like lxml and write a guide for using XPath expressions with requests.
> 
> I was thinking about talking more in detail about stuff like POST and Session and Content-Type for Requests in web scraping, and I could just quickly mention bs4. And you can keep what you currently have for requests, and write more in detail about bs4.

Ok, I looked at your file and I think if you remove the installation part for bs4 and the part 2 on using bs4 that’s probably fine. If you want you could add a link to my wiki entry for that section.

**Comment by tlan4** [comment]
> > > > > Hi, I’ve already made a wiki page on using requests and beautiful soup (see pull request #234 ) so unfortunately this pull request will create a conflict in the wiki.
> > > > 
> > > > 
> > > > What if you focus on the BeautifulSoup part and I can write more about the requests portion?
> > > 
> > > 
> > > I need to use the requests library for my guide to make sense. Maybe you could make your guide on using requests for a topic other than web scraping? Or using a different parser like lxml and write a guide for using XPath expressions with requests.
> > 
> > 
> > I was thinking about talking more in detail about stuff like POST and Session and Content-Type for Requests in web scraping, and I could just quickly mention bs4. And you can keep what you currently have for requests, and write more in detail about bs4.
> 
> Ok, I looked at your file and I think if you remove the installation part for bs4 and the part 2 on using bs4 that’s probably fine. If you want you could add a link to my wiki entry for that section.

Alright sounds good

File: Topics/Tech_Stacks/Python_Requests.md [hunk 1]
```diff
@@ -27,7 +27,28 @@ To determine the parameters needed, you can visit the website using your web bro
...
+  'hl': 'en',     # Set the language to English
+  'q': 'python',  # Keyword to search
+}
+```
+After executign the request, you can check the `status_code` of the request to determine its success. A value of `200` indicates success.
+```python
+if response.status_code != 200:
+  # failure
+```
+A full list of HTTP response codes can be found [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).
```
[the remark this answers is not in the document]
Comment (tlan4) [reply Topics/Tech_Stacks/Python_Requests.md]: Good idea, I expanded on the meaning of the status codes a bit more, but I felt that in most cases you do not have to worry about the specific code, only if it returned 200 or not.

**Comment by tlan4** [comment]
Expanded the status code section and added a section on how to retrieve JSON responses based on the feedback. I also included a new section on User Agents which I felt would be useful as well.

**Review by xingjianll — CHANGES_REQUESTED** [review]
Added comments on parsing.

File: Topics/Tech_Stacks/Python_Requests.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+try:
+  response.raise_for_status()
+except requests.exceptions.HTTPError as e:
+  # handle error here
+```
+
+A full list of HTTP response codes can be found [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).
+
+
+## Parsing
```
Comment (xingjianll) [inline Topics/Tech_Stacks/Python_Requests.md:118]: What is the point of parsing? What does the page look like after we parsed the data? Can you briefly explain beautifulsoup and attach a link to our page that explains the tool?

**Review by xingjianll — CHANGES_REQUESTED** [review]
Added comments for parsing.

File: Topics/Tech_Stacks/Python_Requests.md (hunk 2 above)
Comment (tlan4) [reply Topics/Tech_Stacks/Python_Requests.md:117]: Thanks, I will expand the section and include a link to a more detailed tutorial

**Pull request merged**